src/Compilers/Core/Portable/Collections/SmallDictionary.cs

```diff
@@ -385,6 +393,8 @@ private static AvlNode RightSimple(AvlNode unbalanced)
 
         private static AvlNode LeftComplex(AvlNode unbalanced)
         {
+            RoslynDebug.Assert(unbalanced.Right != null);
```

Comment:
💭 Inconsistent introduction of `is object` and `!= null` assertions in this file